Mage.Sets/src/mage/cards/l/LodestoneBauble.java

```diff
@@ -0,0 +1,123 @@
+ */
+package mage.cards.l;
+
+import java.util.UUID;
+import mage.abilities.Ability;
+import mage.abilities.DelayedTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.delayed.AtTheBeginOfNextUpkeepDelayedTriggeredAbility;
+import mage.abilities.costs.common.SacrificeSourceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.dynamicvalue.common.StaticValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DrawCardTargetEffect;
+import mage.abilities.effects.common.PutOnLibraryTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.common.FilterBasicLandCard;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.common.TargetCardInASingleGraveyard;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author ThomasLerner
+ */
+public class LodestoneBauble extends CardImpl {
+
+    public LodestoneBauble(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{0}");
+        
+
+        // {1}, {T}, Sacrifice Lodestone Bauble: Put up to four target basic land cards from a player's graveyard on top of his or her library in any order. That player draws a card at the beginning of the next turn's upkeep.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new PutOnLibraryTargetEffect(true), new GenericManaCost(1));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new SacrificeSourceCost());
+        ability.addTarget(new TargetCardInASingleGraveyard(0, 4, new FilterBasicLandCard("basic land cards from a player's graveyard")));
+        ability.addEffect(new LodestoneBaubleEffect());
+        this.addAbility(ability);
+
+    }
+
+    public LodestoneBauble(final LodestoneBauble card) {
+        super(card);
+    }
+
+    @Override
+    public LodestoneBauble copy() {
+        return new LodestoneBauble(this);
+    }
+}
+
+class LodestoneBaubleEffect extends OneShotEffect {
+
+    public LodestoneBaubleEffect() {
+        super(Outcome.Detriment);
+        staticText = "Put up to four target basic land cards from a player's graveyard on top of his or her library in any order."
+                + " That player draws a card at the beginning of the next turn's upkeep";
+    }
+
+    public LodestoneBaubleEffect(final LodestoneBaubleEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LodestoneBaubleEffect copy() {
+        return new LodestoneBaubleEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = null;
+        boolean countered = false;
+        UUID targetId = this.getTargetPointer().getFirst(game, source);

```

Comment:
From the gatherer rulings:
>“Up to four” means zero, one, two, three, or four. If you choose zero targets, you still choose a player. That player will draw a card at the beginning of the next turn’s upkeep.

So if nothing is targeted, a player still needs to be chosen. It's probably why this card wasn't already implemented.

Reply:
This is interesting. I think I can do a player choice during the resolution of the ability if no target is selected, but it's a unique implementation so I don't know if that's a good practice.